internal/uvm/cpugroups.go

```diff
@@ -0,0 +1,120 @@
+package uvm
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/hcs"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+const cpuGroupNullID = "00000000-0000-0000-0000-000000000000"
+
+var HV_STATUS_INVALID_CPU_GROUP_STATE = errors.New("The hypervisor could not perform the operation because the CPU group is entering or in an invalid state.")
```

Comment:
I would write this as `ErrHVStatusInvalidCPUGroupState`, that's more idiomatic for Go.

Reply:
Sure thing, I'll add a comment that it corresponds to the `HV_STATUS_INVALID_CPU_GROUP_STATE` internal error